src/foundry/foundry.js/applications/mainMenu.d.ts

```diff
@@ -70,5 +73,5 @@ declare namespace MainMenu {
     onClick: () => void;
   }
 
-  type MenuStructure = { [name: string]: MenuItem };
+  type MenuStructure = Partial<{ [name: string]: MenuItem }>;
```

Comment:
I know that I've been using the `Partial` solution myself for quite a bit but I recently learned that even without that, if you use the option `noUncheckedIndexedAccess`, you will need to check for `undefined`. So I don't think we need the `Partial`.